### PR TITLE
CON-1020: Publish Conclave SDK to Maven Central

### PR DESCRIPTION
**Pull request by bonifazi-r3**
This patch contains the configuration fixes to publish Conclave SDK artifacts to Maven Central

### CODE REVIEW

**Comment by bonifazi-r3** [comment]
@shamsasari Please have a first look and give feedback. I guess now it will be trial and error, not sure if we want to test this at the last stage (i.e. when we trigger the TC release/1.3 build). We can discuss this tomorrow.

File: build.gradle [hunk 1]
```diff
@@ -343,17 +381,34 @@ for (publishedProject in publishedProjects) {
...
+                        developer {
+                            name = 'Conclave Team'
+                            email = "conclave@r3.com"
+                            organization = 'R3 LLC'
+                            organizationUrl = "https://www.r3.com"
+                        }
+                    }
+                    scm {
+                        connection = "scm:git:git://github.com/R3Conclave/conclave-core-sdk.git"
+                        developerConnection = "scm:git:ssh://github.com:R3Conclave/conclave-core-sdk.git"
```
Comment (shamsasari) [inline build.gradle:410]: According to the POM docs, this is only for write access, whatever that means. I think we should remove this one.

File: build.gradle [hunk 2]
```diff
@@ -343,17 +381,34 @@ for (publishedProject in publishedProjects) {
             maven(MavenPublication) {
                 from project.components.java
                 pom {
+                    groupId = project.group
+                    artifactId = project.name
+                    version = project.version
+                    packaging = "jar"
```
Comment (shamsasari) [inline build.gradle:387]: These are all defaults and so there should be no need to specify them.

File: build.gradle [hunk 3]
```diff
@@ -39,9 +39,34 @@ String getConclaveRevision() {
...
+
 ext {
     conclave_revision = getConclaveRevision()
     project.logger.info("Conclave revision: $conclave_revision")
+
+    if (version.endsWith("-SNAPSHOT")) {
+        artifactType = ArtifactType.SNAPSHOT
+    } else if (version.matches(/.+\-RC[0-9]+$/)) {
+        artifactType = ArtifactType.RELEASE_CANDIDATE
+    } else if ((gitBranch == "release/$conclave_version") && ossRHUsername != null) {
```
Comment (shamsasari) [inline build.gradle:60]: The version string should be enough to determine what the version type is, no need to double-check the branch.

File: build.gradle [hunk 4]
```diff
@@ -39,9 +39,34 @@ String getConclaveRevision() {
     return conclave_revision
 }
 
+enum ArtifactType {
```
Comment (shamsasari) [inline build.gradle:42]: I think VersionType is better name. ArtifactType implies we're dealing with different artifacts, whatever that might mean (jar, war, etc).

**Review by shamsasari — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 2 above)
Comment (bonifazi-r3) [reply build.gradle:387]: Ok, I struggled a bit to understand that and thought it was nice to be explicit. Will remove

File: build.gradle (hunk 3 above)
Comment (bonifazi-r3) [reply build.gradle:60]: This is something I had issues with. I.e. when I was playing locally I had a current `release/1.3` branch (by mistake) and I had issues as the build was trying to run the sign task.
I also thought about the possibility where we branch out from the current `release/1.3` and push the changes. In that case, again, we would have issues as the build would try to sign and publish.
The checks above are narrowing the possibility of mistakes like mine

File: build.gradle (hunk 4 above)
Comment (bonifazi-r3) [reply build.gradle:42]: Replaced

File: build.gradle (hunk 1 above)
Comment (bonifazi-r3) [reply build.gradle:410]: [developers](https://central.sonatype.org/publish/requirements/#scm-information)
I understand they are compulsory, but we can put whatever we like there. But if the users do not have access, it does not seem to be a problem

File: build.gradle (hunk 3 above)
Comment (shamsasari) [reply build.gradle:60]: We shouldn't be branching off release/1.3. Why would we do that? Patch releases (e.g. 1.3.1) will continue to be on release/1.3.

I'm not comfortable with this check. It changes the defintion of the version based on the branch and the local setup (i.e. ossRHUsername).

File: build.gradle [hunk 5]
```diff
@@ -343,17 +381,29 @@ for (publishedProject in publishedProjects) {
             maven(MavenPublication) {
                 from project.components.java
                 pom {
+                    name = project.name
```
Comment (shamsasari) [inline build.gradle:384]: Also remove

File: build.gradle [hunk 6]
```diff
@@ -271,6 +285,8 @@ task buildDocs(dependsOn: ['mkdocs'])
 dokkaHtml.shouldRunAfter mkdocs
 // endregion
 
+def projectURL = "https://github.com/R3Conclave/conclave-core-sdk"
```
Comment (shamsasari) [inline build.gradle:288]: We can inline this

**Review by shamsasari — COMMENTED** [review]
Just one minor change, otherwise it's looking good.

**Review by shamsasari — APPROVED** [review]

**Pull request merged**